hermes-mock/src/main/java/pl/allegro/tech/hermes/mock/HermesMockQuery.java

```diff
@@ -123,4 +128,14 @@ public <T> Optional<T> lastAvroMessageAs(String topicName, Schema schema, Class<
         return lastRequest(topicName)
                 .map(req -> hermesMockHelper.deserializeAvro(req, schema, clazz));
     }
+
+    public <T> void resetReceivedAvroRequests(String topicName, Schema schema, Class<T> clazz, Predicate<T> predicate) {
```

Comment:
I'm thinking if `HermesMockQuery` class is good place for these functions. This class responsibility is to `query` Hermes mock, and return some data, not to modify internal state of Hermes mock. 

We have "similar" method `resetReceivedRequest` in `HermesMock` class. Maybe we can place these methods there? What do you think @faderskd?

Reply:
good catch, totally agree :)

Reply:
Sure. Done.